scripts/dts/python-devicetree/src/devicetree/dtlib.py

```diff
@@ -92,6 +93,14 @@ def __init__(self, name: str, parent: Optional['Node'], dt: 'DT'):
         self.parent = parent
         self.dt = dt
 
+        if name.count("@") > 1:
+            dt._parse_error("multiple '@' in node name")
```

Comment:
This could arguably check for `/(@[a-f\d]+)?/` too, since the unit address must be a hex number to comply with this line from section 2.2.1:

> The unit-address must match the first address specified in the reg property of the node.

or is this checked later, and a check like this would be irrelevant here?

Reply:
> This could arguably check for `/(@[a-f\d]+)?/` too, since the unit address must be a hex number to comply with this line from section 2.2.1:

Tricky: that's only the case if the node has a reg property. Without a reg, the spec permits the other characters that are legal in a node name.